includes/wc-admin-update-functions.php

```diff
@@ -316,3 +316,42 @@ function wc_admin_update_290_update_apperance_task_option() {
 function wc_admin_update_290_db_version() {
 	Installer::update_db_version( '2.9.0' );
 }
+
+/**
+ * Use woocommerce_admin_activity_panel_inbox_last_read from the user meta to set wc_admin_notes.is_read col.
+ */
+function wc_admin_update_300_update_is_read_from_last_read() {
+	global $wpdb;
+	$user      = wp_get_current_user();
+	$last_read = get_user_meta( $user->ID, 'woocommerce_admin_activity_panel_inbox_last_read', true );
+
+	// make sure is_read col has been added successfully.
+	$notes_table_cols = $wpdb->get_results( "show columns from {$wpdb->prefix}wc_admin_notes" );
+	$notes_table_cols = array_filter(
+		$notes_table_cols,
+		function( $row ) {
+			// phpcs:ignore
+			return 'is_read' === $row->Field;
+		}
+	);
+	$has_is_read_col  = count( $notes_table_cols ) > 0;
```

Comment:
Not that checking is a bad thing, but curious why we need to check here.  Shouldn't `create_tables` run first, while this runs on a scheduled action later?

Reply:
@joshuatf You're correct. This function gets called after `create_tables`. I'm just trying to be extra careful here as we had a couple of update related issues before.

Reply:
I like the level of precaution here, though I'm very wary of introducing patterns where we're constantly checking to see if something exists.

If we're really hitting these scenarios, it sounds like we have an underlying problem with how migrations are run and I'd prefer to see that addressed one time as opposed to creating these checks each and every time we update the table schema.

Reply:
cc @mattsherman @adrianduffell for a second opinion on this.

I'm a little concerned by this because it sounds like we might be covering up an underlying migration issue.

Reply:
We had a few reports before when we introduced `nonce_action` and `nonce_name`. I believe we still don't know exactly what caused the problem to some customers. 

I agree that this might be a bit too protective. I'm happy to remove it if everyone agrees. I don't have a strong preference for it.

Reply:
I'm happy to remove the defensive code here. I think in this instance, when the new column doesn't exist there would only be an SQL error in the `UPDATE` command. I think this would be logged rather than raise a fatal error in PHP.

As @moon0326 mentioned, there is an underlying issue with the plugin update routine that can can prevent new database columns being added. We saw reports of this in PR https://github.com/woocommerce/woocommerce-admin/pull/6726. I've started an issue around it, which includes more logging to help us identify the cause of these failures https://github.com/woocommerce/woocommerce-admin/issues/7953.

Reply:
Thank you @adrianduffell for the comment 👍 

@joshuatf Removed the check in https://github.com/woocommerce/woocommerce-admin/commit/5fb2d6f790907e35eb5226d934ac08423604276b